src/olympia/amo/management/commands/check_locales_completion_rate.py

```diff
@@ -0,0 +1,175 @@
+        )
+
+    def handle(self, *args, **options):
+        locales_below_threshold, locales_above_threshold = (
+            self.find_locales_below_and_above_threshold()
+        )
+        self.warn_about(
+            locales_below_threshold=locales_below_threshold,
+            locales_above_threshold=locales_above_threshold,
+        )
+
+    def find_locales_below_and_above_threshold(self):
+        locales_below_threshold = set()
+        locales_above_threshold = set()
+        seen_locales = defaultdict(int)
+        for project in self.PONTOON_PROJECTS:
+            data = self.fetch_data(project)
+            for locale_data in data.get('project', {}).get('localizations', []):
+                locale = self.get_locale(locale_data)
+                seen_locales[locale] += 1
+                completion = self.get_completion(locale_data)
+                # If we see a locale below treshold, immediately add it to the
+                # relevant set.
+                if completion < self.COMPLETION_THRESHOLD:
+                    self.stdout.write(f'❌ {locale} is below threshold ({completion}%)')
+                    locales_below_threshold.add(locale)
+                # If we see a language above threshold but not already enabled
+                # in production, add it to the relevant set.
+                elif locale not in settings.AMO_LANGUAGES:
+                    self.stdout.write(f'✅ {locale} is above threshold ({completion}%)')
+                    locales_above_threshold.add(locale)
+        # Add to locales below threshold locales we only saw in one project.
+        # Use the ones we already have enabled in production, in case somehow
+        # a locale would disappear completely from pontoon and still be enabled
+        # on our side.
+        for locale in settings.AMO_LANGUAGES:
+            if locale != settings.LANGUAGE_CODE and seen_locales[locale] != len(
+                self.PONTOON_PROJECTS
+            ):
+                self.stdout.write(
+                    f'❌ {locale} is only in {seen_locales[locale]} project(s) !'
+                )
+                locales_below_threshold.add(locale)
+        # Somewhat inefficient but good enough check to remove any locales that
+        # both below and above threshold because of differences between the
+        # projects.
+        for locale in locales_above_threshold:
+            if locale in locales_below_threshold:
+                self.stdout.write(f'❌ {locale} is in both sets !')
+                locales_above_threshold.remove(locale)
+        return locales_below_threshold, locales_above_threshold
+
+    def fetch_data(self, project):
+        q = re.sub(r'\s', '', self.QUERY_TEMPLATE % {'project': project})
+        self.stdout.write(f'Calling pontoon with {q}')
+        response = requests.get(self.PONTOON_API, {'query': q})
+        response.raise_for_status()
+        data = response.json().get('data', {})
```

Comment:
Graphql apis generally return errors in the response and usually response 200 even with errors.. we might need to check if pontoon might be returning an "error" object in the response.

Reply:
I had 400 for incorrect fields. For instance, this 400s:
```
curl -gi 'https://pontoon.mozilla.org/graphql?query={projecterrorpleaseignore(slug:"amo-frontend"){name,localizations{locale{code,name}totalStrings,approvedStrings}}}' 
```

Reply:
Just wondering if it returns an error property in data. GraphQL apis are weird and can fail both with status or property. Depends on how pontoon implemented it but might be worth double checking.

Reply:
It does return `errors` but it would raise before we parse since it returns a proper error code.